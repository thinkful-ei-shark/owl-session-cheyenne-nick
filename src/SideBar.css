.side-bar {
    border: 1px solid black;
}