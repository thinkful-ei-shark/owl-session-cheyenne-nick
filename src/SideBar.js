import React, { Component } from 'react';
import SideBarPerson from './SideBarPerson';
import './SideBar.css';

class SideBar extends Component {
  findById(id) {
    return this.props.participants.find(participant => participant.id === id);
  }

  render() {
    const eventShoutouts = {
      'thumbs-up': ' gave a thumbs up 👍',
      'thumbs-down': ' gave a thumbs down 👎',
      'raise-hand': ' raised their hand ✋',
      'clap': ' clapped 👏',
      'join': ' joined',
      'leave': ' left',
      'join-stage': ' joined the stage',
      'leave-stage': ' left the stage'
    };

    const chatEvents = this.props.chatEvents.map(event => {
      const currentParticipant = this.findById(event.participantId);
      if (event.type === "message") {
        return (
          <SideBarPerson key={event.participantId + event.timestamp}
            name={currentParticipant.name} time={event.time} 
            messages={[event.message]} avatar={currentParticipant.avatar}
            isInChat={true} />
        );
      } else {
        return (
          <p key={event.participantId + event.timestamp} className="chat-event item">
            <span className="user-name">{currentParticipant.name}</span>
            {eventShoutouts[event.type]}
          </p>
        );
      }
    });

    let participantsList = this.props.participants.map(participant => {
      if (participant.inSession) {
        return (
          <SideBarPerson key={participant.id} name={participant.name}
            avatar={[participant.avatar]} onStage={participant.onStage}
            inSession={participant.inSession} isInChat={false} />
        );
      }
    });
    // Loop for participants not in session
    participantsList.concat(this.props.participants.map(participant => {
      if (!participant.inSession) {
        return (
          <SideBarPerson key={participant.id} name={participant.name}
            avatar={[participant.avatar]} onStage={participant.onStage}
            inSession={participant.inSession} isInChat={false} />
        );
      }
    }));


    return (
      <section className="side-bar item group-column">
        <div className="actions-header group-row">
          <button type="button" className="item">Chat</button>
          <button type="button" className="item">Participants {this.props.participants.length}</button>
          <button type="button" className="item">⚙️</button>
          <button type="button" className="item">🔗</button>
          <button type="button" className="item">?</button>
        </div>

        <ul className="chat-log item">
          {chatEvents}
        </ul>

        <ul className="participant-list hidden item">
          {participantsList}
        </ul>

        <textarea placeholder="Chat"></textarea>
      </section>
    );
  }
}

export default SideBar;