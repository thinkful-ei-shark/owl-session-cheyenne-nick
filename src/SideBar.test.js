import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import store from './STORE'
import SideBar from './SideBar'

describe('<SideBar />', () => {
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<SideBar participants={store.participants} chatEvents={store.chatEvents} />, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {

      const sideBar = renderer
          .create(<SideBar participants={store.participants} chatEvents={store.chatEvents} />)
          .toJSON();
      
      expect(sideBar).toMatchSnapshot();
    });
});
