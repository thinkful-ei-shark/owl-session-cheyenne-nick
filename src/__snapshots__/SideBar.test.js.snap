// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> renders the UI as expected 1`] = `
<section
  className="side-bar item group-column"
>
  <div
    className="actions-header group-row"
  >
    <button
      className="item"
      type="button"
    >
      Chat
    </button>
    <button
      className="item"
      type="button"
    >
      Participants 
      5
    </button>
    <button
      className="item"
      type="button"
    >
      ‚öôÔ∏è
    </button>
    <button
      className="item"
      type="button"
    >
      üîó
    </button>
    <button
      className="item"
      type="button"
    >
      ?
    </button>
  </div>
  <ul
    className="chat-log item"
  >
    <li
      className="side-bar-person group-row"
    >
      <img
        alt="Axe Kubicka"
        className="profile-pic item"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <div
        className="person-data group-vertical item"
      >
        <div
          className="group-row"
        >
          <h3
            className="item"
          >
            this.props.name
          </h3>
          <p
            className="item"
          >
            this.props.time
          </p>
        </div>
        <div
          className="person-data-content group-vertical"
        >
          <p
            className="item"
          >
            Hello world
          </p>
        </div>
      </div>
    </li>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Caty Flucker
      </span>
       gave a thumbs up üëç
    </p>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Frank Runciman
      </span>
       gave a thumbs down üëé
    </p>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Koren Templeton
      </span>
       raised their hand ‚úã
    </p>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Koren Templeton
      </span>
       clapped üëè
    </p>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Koren Templeton
      </span>
       joined
    </p>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Ashla Attwool
      </span>
       left
    </p>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Axe Kubicka
      </span>
       joined the stage
    </p>
    <p
      className="chat-event item"
    >
      <span
        className="user-name"
      >
        Axe Kubicka
      </span>
       left the stage
    </p>
  </ul>
  <ul
    className="participant-list hidden item"
  >
    <li
      className="side-bar-person group-row"
    >
      <img
        alt="Koren Templeton"
        className="profile-pic item"
        src={
          Array [
            "https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1",
          ]
        }
      />
      <div
        className="person-data group-vertical item"
      >
        <h3
          className="item"
        >
          this.props.name
        </h3>
        <div
          className="person-data-content group-row"
        >
          <div
            className="circle item"
          />
          <p
            className="stage-status"
          >
            on stage
          </p>
          <button
            className="boot"
          >
            ‚Ä¢ Boot
          </button>
        </div>
      </div>
    </li>
    <li
      className="side-bar-person group-row"
    >
      <img
        alt="Caty Flucker"
        className="profile-pic item"
        src={
          Array [
            "https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1",
          ]
        }
      />
      <div
        className="person-data group-vertical item"
      >
        <h3
          className="item"
        >
          this.props.name
        </h3>
        <div
          className="person-data-content group-row"
        >
          <div
            className="circle item"
          />
          
          <button
            className="boot"
          >
            ‚Ä¢ Boot
          </button>
        </div>
      </div>
    </li>
    <li
      className="side-bar-person group-row"
    >
      <img
        alt="Frank Runciman"
        className="profile-pic item"
        src={
          Array [
            "https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1",
          ]
        }
      />
      <div
        className="person-data group-vertical item"
      >
        <h3
          className="item"
        >
          this.props.name
        </h3>
        <div
          className="person-data-content group-row"
        >
          <div
            className="circle item"
          />
          
          <button
            className="boot"
          >
            ‚Ä¢ Boot
          </button>
        </div>
      </div>
    </li>
    <li
      className="side-bar-person group-row"
    >
      <img
        alt="Ashla Attwool"
        className="profile-pic item"
        src={
          Array [
            "https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1",
          ]
        }
      />
      <div
        className="person-data group-vertical item"
      >
        <h3
          className="item"
        >
          this.props.name
        </h3>
        <div
          className="person-data-content group-row"
        >
          <div
            className="circle item"
          />
          <p
            className="stage-status"
          >
            on stage
          </p>
          <button
            className="boot"
          >
            ‚Ä¢ Boot
          </button>
        </div>
      </div>
    </li>
  </ul>
  <textarea
    placeholder="Chat"
  />
</section>
`;
